File: 프로그래머스/unrated/133502. 햄버거 만들기/햄버거 만들기.py
def solution(ingredient):
    answer = 0
    
    buger = [1, 2, 3, 1]
    idx = 0
    
    while idx <= len(ingredient) - 4:
        if ingredient[idx] != 1:
            idx += 1 
        else:
            tmp = ingredient[idx : idx + 4]
            if tmp == buger:
                answer += 1
                del ingredient[idx : idx + 4]
                if idx < 4:
                    idx = 0
                else:
                    idx -= 4
                    
            else:
                idx += 1
        
    return answer

File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.js
function solution(ingredient) {
    var answer = 0;
    const bugger_order = [1, 2, 3, 1];
    
    let idx = 0
    while (idx <= ingredient.length - 4) {
        if (ingredient[idx] !== 1) idx++;
        else {
            const order = ingredient.slice(idx, idx + 4);
            const order_filter = order.filter((ele, i) => ele !== bugger_order[i]).length;
            if (!order_filter) {
                ingredient.splice(idx, 4);
                answer++;
                idx = idx < 4 ? 0 : idx - 2
            } else idx++;  
        }
    }
    
    return answer;
}